### PR TITLE
feat(example): add nx example

### PR DESCRIPTION
**Pull request by tknickman**
Adds a new Next.js example with Nx. 

Supports single button deploy for an Nx Next.js app on Vercel.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: examples/with-nx/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  },
+  "private": true,
+  "dependencies": {
+    "@nrwl/next": "14.6.3",
+    "core-js": "^3.6.5",
+    "next": "12.2.5",
+    "react": "18.2.0",
+    "react-dom": "18.2.0",
+    "regenerator-runtime": "0.13.7",
+    "tslib": "^2.3.0"
```
Comment (ijjk) [inline examples/with-nx/package.json:15]: Are `core-js`, `regenerator-runtime`, and `tslib` needed as dependencies?

File: examples/with-nx/apps/app/pages/index.module.css [hunk 2]
```diff
@@ -0,0 +1,2 @@
+.page {
```
Comment (ijjk) [inline examples/with-nx/apps/app/pages/index.module.css:1]: Is this meant to be empty as a placeholder?

File: examples/with-nx/apps/app/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+declare module '*.svg' {
+  const content: any;
+  export const ReactComponent: any;
```
Comment (ijjk) [inline examples/with-nx/apps/app/index.d.ts:4]: Should this type export `ReactComponent` if `svgr: false` is the default in `next.config.js`?

File: examples/with-nx/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
+# Example app using Nx
+
+**NOTE:** When using Nx on Vercel, ensure you are using `nx@14.6.2` or above.
+
+This example was created using [Nx](https://nx.dev).
+
```
Comment (ijjk) [inline examples/with-nx/README.md:6]: Do we have a deployed example we can add here?

File: examples/with-nx/package.json (hunk 1 above)
Comment (tknickman) [reply examples/with-nx/package.json]: Good catch, this was generated from a boilerplate `npx create-nx-workspace` so some of this looks to be unused for the simpler cases.

File: examples/with-nx/apps/app/pages/index.module.css (hunk 2 above)
Comment (tknickman) [reply examples/with-nx/apps/app/pages/index.module.css]: This was definitely leftover from the generator. Removed 👍🏼

File: examples/with-nx/apps/app/index.d.ts (hunk 3 above)
Comment (tknickman) [reply examples/with-nx/apps/app/index.d.ts]: Yea I took a closer look here and just removed this completely - another remnant of the boilerplate.

File: examples/with-nx/README.md (hunk 4 above)
Comment (tknickman) [reply examples/with-nx/README.md:6]: We don't right now but I can add one.

File: examples/with-nx/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+If using `@nrwl/nx-cloud@14.6.0` or above
+
+1. Set `NX_CACHE_DIRECTORY=/tmp/nx-cache`
+
+If using `@nrwl/nx-cloud@14.5.0` or below
+
+1. Set `NX_CACHE_DIRECTORY=/tmp/nx-cache`
+2. Set the `cacheDirectory` option for the `@nrwl/nx-cloud` runner in your `nx.json` to match the value of the `NX_CACHE_DIRECTORY` environment variable:
+
+```json
```
Comment (gaspar09) [inline examples/with-nx/README.md:80]: You can use `jsonc` here

**Pull request closed without merging**